src/executables/util_cs.c

```diff
@@ -4544,18 +4545,114 @@ flashback (UTIL_FUNCTION_ARG * arg)
 int
 memmon (UTIL_FUNCTION_ARG * arg)
 {
+#if defined(CS_MODE)
   UTIL_ARG_MAP *arg_map = arg->arg_map;
-  char *module;
-  bool transaction, show_all;
+  char er_msg_file[PATH_MAX];
+  const char *database_name;
+  char *module, *stop;
+  bool transaction, show_all, need_shutdown;
   int tran_count;
+  int module_index = MMON_MODULE_LAST;
 
   module = utility_get_option_string_value (arg_map, MEMMON_MODULE_S, 0);
   transaction = utility_get_option_bool_value (arg_map, MEMMON_TRANSACTION_S);
   tran_count = utility_get_option_int_value (arg_map, MEMMON_TRAN_COUNT_S);
   show_all = utility_get_option_bool_value (arg_map, MEMMON_SHOW_ALL_S);
 
+  database_name = utility_get_option_string_value (arg_map, OPTION_STRING_TABLE, 0);
+  if (database_name == NULL)
+    {
+      goto print_memmon_usage;
+    }
+
+  if (check_database_name (database_name))
+    {
+      goto error_exit;
+    }
+
+  if ((!transaction) && tran_count)
+    {
+      PRINT_AND_LOG_ERR_MSG (msgcat_message
+			     (MSGCAT_CATALOG_UTILS, MSGCAT_UTIL_SET_MEMMON, MEMMON_MSG_INVALID_TRAN_COUNT_OPTION));
+      goto error_exit;
+    }
+
+  if (tran_count < 0)
+    {
+      goto print_memmon_usage;
+    }
+
+  if (show_all && (transaction || module))
+    {
+      PRINT_AND_LOG_ERR_MSG (msgcat_message
+			     (MSGCAT_CATALOG_UTILS, MSGCAT_UTIL_SET_MEMMON, MEMMON_MSG_EXCLUSIVE_OPTION));
+      goto error_exit;
+    }
+
+  module_index = strtol (module, &stop, 10);
+  if (module_index)
+    {
+      if (module_index < 0 || module_index > MMON_MODULE_LAST)
+	{
+	  PRINT_AND_LOG_ERR_MSG (msgcat_message
+				 (MSGCAT_CATALOG_UTILS, MSGCAT_UTIL_SET_MEMMON, MEMMON_MSG_NO_MATCHING_MODULE), module);
+	  goto error_exit;
+	}
+    }
+  else
+    {
+      /* module == NULL or module with module name */
+      // TODO: convert module name to module index
+    }
+
+  /* error message log file */
+  snprintf (er_msg_file, sizeof (er_msg_file) - 1, "%s_%s.err", database_name, arg->command_name);
+  er_init (er_msg_file, ER_NEVER_EXIT);
+
+  if (db_restart (arg->command_name, TRUE, database_name))
+    {
+      PRINT_AND_LOG_ERR_MSG ("%s: %s. \n\n", arg->command_name, db_error_string (3));
+      goto error_exit;
+    }
+  need_shutdown = true;
+
+  if (!prm_get_bool_value (PRM_ID_MEMORY_MONITORING))
+    {
+      PRINT_AND_LOG_ERR_MSG (msgcat_message (MSGCAT_CATALOG_UTILS, MSGCAT_UTIL_SET_MEMMON, MEMMON_MSG_NOT_SUPPORTED));
+      goto error_exit;
+    }
+
   // XXX: for test, it will removed at main implementation
   fprintf (stdout, "memmon utility: -m %s, -t %s, -c %d, -a %s\n", (module ? module : "NULL"),
 	   (transaction ? "true" : "false"), tran_count, (show_all ? "true" : "false"));
+
+  if (need_shutdown)
```

Comment:
It's always true here.

Reply:
Are there any need for checking corner case? ex) `assert(need_shutdown == true)`

Reply:
At your choice. I'd not do it, though.